### PR TITLE
Add vggface2 dataset

### PR DESCRIPTION
**Pull request by jgbradley1**
Closes issue #1139.

There was a [PR](https://github.com/pytorch/vision/pull/1193) for this last year but it never got approved and finalized. Since the design of the Dataset classes has changed quite a bit since then, I thought I would submit this more updated version for review.

Notable changes:
* The dataset class uses bbox and landmark annotations that are available on the VGGFace2 website. From what I can tell, these are the only annotations that are available for all identities in the dataset. Other annotation information is available but only for a subset of face identities.
* No preprocessing of the dataset is performed. This option is left up to the user who can freely define `transform`, `target_type` and `target_transform` in order to make whatever changes they want.

cc @fmassa @dakshjotwani

### CODE REVIEW

**Comment by dakshjotwani** [comment]
Thanks for bringing this PR back to life :) 

Can we add test cases for `VGGFace2` similar to the imagenet dataset? I remember I was about to do them for my PR before we forgot about it.

File: torchvision/datasets/vggface2.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    def __init__(
+        self,
+        root: str,
+        split: str = "train",
+        target_type: Union[List[str], str] = "bbox",
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+        download: bool = False,
+    ) -> None:
+        import pandas
```
Comment (dakshjotwani) [inline torchvision/datasets/vggface2.py:69]: Do we really need to use pandas? I'm not sure if its worth importing a new library just for `concat`. We should be fine with just a dict here

**Comment by jgbradley1** [comment]
Unit tests have been added and tested.

**Comment by jgbradley1** [comment]
This PR is as ready-to-go as it gets I think. Unless someone has an issue with return types or documentation suggestions, I don't have anything else to add.

**Comment by jgbradley1** [comment]
Before more work is done on this PR, we should wait to see if the dataset is made available again. Looks like the dataset has been taken down for [almost a month](https://github.com/ox-vgg/vgg_face2/issues/52).

VGGFace2 citation count: [912](https://scholar.google.com/scholar?cites=17598700583298216808&as_sdt=20000005&sciodt=0,21&hl=en)

**Comment by yassineAlouini** [comment]
This is a new dataset request and will be added to the new dataset API once it is stable enough. In the meantime, we will close the current PR (don't worry, the code will be reused at some point in the new dataset design to integrate it). @pmeier

**Comment by pmeier** [comment]
To complement @yassineAlouini's comment:

I'm not sure how I missed this PR since it seems complete. I'm very sorry about that @jgbradley1. Unfortunately, we are moving to a new datasets API in the near future and thus, we will no longer accept PRs that add new datasets to the old API. There are two options now:

1. Given that you already put in the work in this PR, we would happily accept a new PR adding this to the new API. You can refer to [this guide]() on how to do that.
2. Someone else could take this over and build on what you have done. In this case I will make sure to credit you properly.

How would you like to go about it @jgbradley1?

**Pull request closed without merging**